avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -691,9 +691,6 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
       case STRING:
         if (enhancedSchemaSupport && schema.parameters() != null
             && schema.parameters().containsKey(AVRO_TYPE_ENUM)) {
-          String enumSchemaName = schema.parameters().get(AVRO_TYPE_ENUM);
-          org.apache.avro.Schema cachedEnum = schemaMap.get(enumSchemaName);
-          if (cachedEnum == null) {
```

Comment:
I refer to the `STRING` switch case as the "top-level" schema type since it's the type of the input `org.apache.kafka.connect.data.Schema`. When it switch matches the schema at that level, it can either be an Avro String schema type **or** an Avro enumeration type. In either case, at the "top-level" or "outside level", it's still a `STRING` type, according to this code.

This code will still process the enum message type, that hasn't been removed. Only the caching of that constructed enum type has been removed. 

To put this change in another way, I can pose this question to maybe help me understand it better: why do we need to cache the `baseSchema` (previously line 708) constructed by discovering that the `schema.parameters()` contains `AVRO_TYPE_ENUM` **if it will be cached on line 853 regardless**? The next time we come across the input `schema`, our Hashmap cache will have matched the object exactly, including the fact that it contains `AVRO_TYPE_ENUM` as a parameter. This means we would never retrieve from the cache the constructed enum schema because we would have got a cache hit on line 649. I understand that maybe originally the idea was to cache on the Avro Enum type, maybe optimally, but that was done using the String name of the input schema, something we are suggesting not do anymore.